korangar/src/world/map/mod.rs

```diff
@@ -265,10 +265,27 @@ impl Map {
 
     #[cfg_attr(feature = "debug", korangar_debug::profile)]
     pub fn render_entities(&self, instructions: &mut Vec<EntityInstruction>, entities: &[Entity], camera: &dyn Camera, include_self: bool) {
-        entities
-            .iter()
-            .skip(!include_self as usize)
-            .for_each(|entity| entity.render(instructions, camera));
+        fn get_distance(camera: &dyn Camera, entity: &Entity) -> f32 {
+            camera.distance_to(entity.get_position())
+        }
+        let mut entity_order: Vec<usize> = (0..entities.len()).collect();
+
+        entity_order.sort_by(|index_a, index_b| {
+            let dist_a = get_distance(camera, &entities[*index_a]);
+            let dist_b = get_distance(camera, &entities[*index_b]);
+            dist_b.partial_cmp(&dist_a).unwrap()
+        });
+
+        entity_order.iter().for_each(|&index| {
+            let is_self = index == 0;
```

Comment:
"self" has a pretty well defined meaning in Rust. We should better use "is_player_entity" or somthing similar.